docs/adding_a_feature.md

```diff
@@ -0,0 +1,211 @@
+	c := buildDefaultCluster(t)
+	c.Spec.UpdatePolicy = fi.String("not-a-real-value")
+	expectErrorFromValidate(t, c, "UpdatePolicy")
+}
+```
+
+
+And we should test the nodeup tag building:
+
+```
+func TestBuildTags_UpdatePolicy_Nil(t *testing.T) {
+	c := &api.Cluster{
+		Spec: api.ClusterSpec{
+			UpdatePolicy: nil,
+		},
+	}
+
+	tags, err := buildClusterTags(c)
+	if err != nil {
+		t.Fatalf("buildTags error: %v", err)
+	}
+
+	nodeUpTags, err := buildNodeupTags(api.InstanceGroupRoleNode, c, tags)
+	if err != nil {
+		t.Fatalf("buildNodeupTags error: %v", err)
+	}
+
+	if !stringSliceContains(nodeUpTags, "_automatic_upgrades") {
+		t.Fatalf("nodeUpTag _automatic_upgrades not found")
+	}
+}
+
+func TestBuildTags_UpdatePolicy_External(t *testing.T) {
+	c := &api.Cluster{
+		Spec: api.ClusterSpec{
+			UpdatePolicy: fi.String("external"),
+		},
+	}
+
+	tags, err := buildClusterTags(c)
+	if err != nil {
+		t.Fatalf("buildTags error: %v", err)
+	}
+
+	nodeUpTags, err := buildNodeupTags(api.InstanceGroupRoleNode, c, tags)
+	if err != nil {
+		t.Fatalf("buildNodeupTags error: %v", err)
+	}
+
+	if stringSliceContains(nodeUpTags, "_automatic_upgrades") {
+		t.Fatalf("nodeUpTag _automatic_upgrades found unexpectedly")
+	}
+}
+```
+
+## Documentation
+
+Add some documentation on your new field:
+
+```
+## UpdatePolicy
+
+Cluster.Spec.UpdatePolicy
+
+Values:
+
+* `external` do not enable automatic software updates
+
+* unset means to use the default policy, which is currently to apply OS security updates unless they require a reboot
+```
+
+## Testing
+
+
+You can `make` and run `kops` locally.  But `nodeup` is pulled from an S3 bucket.
+
+To rapidly test a nodeup change, you can build it, scp it to a running machine, and
+run it over SSH with the output viewable locally:
+
+`make push-aws-run TARGET=admin@<publicip>`
+
+
+For more complete testing though, you will likely want to do a private build of
+nodeup and launch a cluster from scratch.
+
+To do this, you can repoint the nodeup source url by setting the `NODEUP_URL` env var,
+and then push nodeup using:
+
+
+```
+export S3_BUCKET_NAME=<yourbucketname>
+make upload S3_BUCKET=s3://${S3_BUCKET_NAME}
+
+export NODEUP_URL=https://${S3_BUCKET_NAME}.s3.amazonaws.com/nodeup/nodeup-1.3.tar.gz
+
+kops create cluster <clustername> --zones us-east-1b
+...
+```
+
+## Using the feature
+
+Users would simply `kops edit cluster`, and add a value like:
+```
+  spec:
+    updatePolicy: external
+```
+
+Then `kops update cluster --yes` would create the new NodeUpConfig, which is included in the instance startup script
+and thus requires a new LaunchConfiguration, and thus a `kops rolling update`.  We're working on changing settings
+without requiring a reboot, but likely for this particular setting it isn't the sort of thing you need to change
+very often.
+
+## Other steps
+
+* We could also create a CLI flag on `create cluster`.  This doesn't seem worth it in this case; this is a relatively advanced option
```

Comment:
Yah but it is tough for programs to use `kubectl edit cluster`, so we have to code editing an s3 file.  Not rocket science, but do you mind me committing a cli switch?